### PR TITLE
[thing] Added default channel types for electrical energy

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added default channel types for electrical energy (electric power, electric current, electric potential and electrical energy)

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java [hunk 1]
```diff
@@ -347,13 +355,56 @@ public class DefaultSystemChannelTypeProvider implements ChannelTypeProvider {
...
+
+    /**
+     * Electric-potential: system wide {@link ChannelType} which shows the electric potential
+     */
+    public static final ChannelType SYSTEM_ELECTRIC_POTENTIAL = ChannelTypeBuilder
+            .state(SYSTEM_CHANNEL_TYPE_UID_ELECTRIC_POTENTIAL, "Electric Potential", "Number:ElectricPotential")
+            .withDescription("Current electric potential").withCategory("Energy")
+            .withStateDescriptionFragment(
+                    StateDescriptionFragmentBuilder.create().withReadOnly(true).withPattern("%.1f %unit%").build())
+            .withTags(List.of("Measurement", "Voltage")).build();
```
Comment (J-N-K) [inline bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java:388]: Are you sure that this is a good naming? Usually it'll probably be a potential difference (aka voltage), not na absolute potential.

Comment (cweitkamp) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java:388]: No, I am not. I tried to create a relation to the dimension to be used with the ItemType and a cluster for electrical Channels. In general we can name all of them just Current, Power, Voltage and Energy - without the prefix.

Comment (J-N-K) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java:388]: IMO that would be better from the user perspective.

Comment (cweitkamp) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java:388]: Me too.

But there is one problem. The channel type `power` already is user for turning devices on and off:

https://github.com/openhab/openhab-core/blob/2779c0fe0fd36872c6cdcc1cab006d3c774b2d90/bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java#L70

Comment (J-N-K) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java:388]: Maybe keep the "Electric" prefix then. There will probably be no clash for "voltage" but "current" could also be related to water or gas.

Comment (cweitkamp) [reply bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/DefaultSystemChannelTypeProvider.java:388]: Okay. Then I will rename just electric potential to electric voltage and keep the rest.

**Review by J-N-K — APPROVED** [review]

**Pull request merged**